Content.Shared/Weapons/Ranged/Systems/SharedGunSystem.Ballistic.cs

```diff
@@ -252,6 +243,11 @@ private void OnBallisticTakeAmmo(EntityUid uid, BallisticAmmoProviderComponent c
                 component.Entities.RemoveAt(component.Entities.Count - 1);
                 DirtyField(uid, component, nameof(BallisticAmmoProviderComponent.Entities));
                 Containers.Remove(entity, component.Container);
+
+                if (TryComp<BallisticAmmoSelfRefillerComponent>(uid, out var refiller))
+                {
+                    PauseSelfRefill((uid, refiller));
+                }
```

Comment:
Shouldn't this also occur in the `else if (component.UnspawnedCount > 0)` right below as well?